csci-499-fa24/Team6
# Make add macros a pop up

### **Issue / Feature**
Implement a popup modal for the "Add Macros" functionality.

---

### **Description**
This PR introduces a popup modal that appears when the "Add Macros" button is clicked. The modal allows users to input and save consumed macros while also providing a "Cancel" option to close the modal. This change enhances the user experience by presenting the functionality in a focused and visually appealing way.

---

### **What's in this change?**
- Added a centered modal (`.addMacrosSection`) with fixed positioning.
- Included an overlay (`.overlay`) to dim the background when the modal is active.
- "Add Macros" and "Cancel" buttons for interaction.
- Styled the modal for alignment, spacing, and visual hierarchy.
- Implemented logic to toggle the visibility of the modal.

---

### **What's left to do?**
- [ ] Refactor shared modal logic for reusability in other features.
- [ ] Cross-browser testing for consistent rendering.
- [ ] Accessibility improvements (e.g., keyboard navigation, screen reader support).

---

### **Testing changes**
1. Click the "Add Macros" button to open the modal.
2. Verify that the modal appears centered with a dimmed background.
3. Enter macro values and click "Add Macros" to ensure data is handled correctly.
4. Click "Cancel" to verify that the modal closes without changes.

<img width="1470" alt="image" src="https://github.com/user-attachments/assets/0410f2c4-2937-4834-b26f-3fd1547467df">
